### PR TITLE
Use target triples to describe action binaries

### PR DESCRIPTION
**Pull request by maiste**
Hi
This PR changes the name of the binary targets to make them use _triple targets_ instead.

### CODE REVIEW

File: .github/workflows/binaries.yml [hunk 1]
```diff
@@ -11,16 +11,16 @@ jobs:
...
-            name: mac-arm
+            name: aarch64-apple-darwin
             installable: .#
           - os: ubuntu-22.04
-            name: linux-intel
+            name: x86_64-unknown-linux-gnu
             installable: .#
           - os: ubuntu-22.04
-            name: linux-intel-static
+            name: x86_64-unknow-linux-musl
```
Comment (emillon) [inline .github/workflows/binaries.yml:23]: typo here

File: .github/workflows/binaries.yml [hunk 2]
```diff
@@ -11,16 +11,16 @@ jobs:
...
         include:
           - os: macos-13
-            name: mac-intel
+            name: x86_64-apple-darwin
             installable: .#
           - os: macos-14
-            name: mac-arm
+            name: aarch64-apple-darwin
             installable: .#
           - os: ubuntu-22.04
```
Comment (emillon) [inline .github/workflows/binaries.yml:19]: I thought we got rid of that one, maybe that was only on the toolchains branch

Comment (maiste) [reply .github/workflows/binaries.yml:19]: Do you want me to remove this target or we leave it here?

Comment (emillon) [reply .github/workflows/binaries.yml:19]: can you do it in a separate PR so we don't forget?

Comment (maiste) [reply .github/workflows/binaries.yml:19]: Sure, no problem :+1: 
Done in #10683

**Pull request merged**